### DISCUSSION TITLE
0.30.0

### DISCUSSION THREAD

**Discussion by AmrDeveloper** [AmrDeveloper/GQL#128] [Announcements]
![Screenshot 2024-11-08 at 19 19 09](https://github.com/user-attachments/assets/ac493d9c-9400-42cb-bc59-825223a14e9f)

- Implement Composite type.
- Implement Member access expression for Composite type.
- Replace `atty` with std is_terminal.
- Improve order of iterating over composite type members.
- Fix comparions operators for String, Date, Time and DateTime

<hr /><em>This discussion was created from the release <a href='https://github.com/AmrDeveloper/GQL/releases/tag/0.30.0'>0.30.0</a>.</em>

**Comment by jorgeaponte** [comment]
Hi @AmrDeveloper, 
I was trying to create a query to get all the commits after a specific date with no luck.
I tried the following:
```sql
SELECT * FROM commits WHERE datetime > '2024-09-11 01:01:01'
SELECT * FROM commits WHERE datetime > CAST('2024-09-11' AS DATE)
SELECT * FROM commits WHERE datetime >  Date('2024-09-11')
SELECT * FROM commits WHERE datetime @> DATERANGE ('2024-09-11', '2024-11-11')
SELECT * FROM commits WHERE datetime > MAKEDATE(2024, 300) 
```
What is the proper way to do it in this version?
Could you give me a hand with this?

**Comment by AmrDeveloper** [comment]
Hello @jorgeaponte,

There was parsing error in current unreleased parser and datetime type is `DateTime` so you should compare it with DateTime, Please Check 0.31.0

```sql
- SELECT * FROM commits WHERE datetime > '2024-09-11 01:01:01' // Working
- SELECT * FROM commits WHERE datetime > CAST('2024-09-11' AS DATE) // Not fully supported yet, will supported next release
- SELECT * FROM commits WHERE datetime >  Date('2024-09-11') // Can't compare Date vs DateTime
- SELECT * FROM commits WHERE TSRANGE ('2024-09-11 01:01:01', '2024-11-11 01:01:01') @> datetime // Working but contains expect left hand side to be range
- SELECT * FROM commits WHERE datetime <@ TSRANGE ('2024-09-11 01:01:01', '2024-11-11 01:01:01') // Will be valid in 0.32.0
```

Thank you for reporting
